pulsar-client/src/test/java/org/apache/pulsar/client/impl/ConsumerBuilderImplTest.java

```diff
@@ -346,6 +349,36 @@ public void testNullDeadLetterPolicy() {
         verify(consumerBuilderImpl.getConf()).setDeadLetterPolicy(null);
     }
 
+    @Test
+    public void testNonNullDeadLetterPolicy() {
+        PulsarClientImpl client = mock(PulsarClientImpl.class);
+        ConsumerConfigurationData consumerConfigurationData = mock(ConsumerConfigurationData.class);
+        doCallRealMethod().when(consumerConfigurationData).setDeadLetterPolicy(any());
+        when(consumerConfigurationData.getDeadLetterPolicy()).thenCallRealMethod();
+        ConsumerBuilderImpl<byte[]> consumerBuilder = new ConsumerBuilderImpl(client, consumerConfigurationData, Schema.BYTES);
+
+        consumerBuilder.deadLetterPolicy(DeadLetterPolicy.builder()
+                .retryLetterTopic("new-retry")
+                .initialSubscriptionName("new-dlq-sub")
+                .deadLetterTopic("new-dlq")
+                .maxRedeliverCount(2)
+                .deadLetterBatchingEnabled(false)
+                .deadLetterChunkingEnabled(false)
+                .retryLetterBatchingEnabled(true)
+                .retryLetterChunkingEnabled(true)
```

Comment:
We couldn't enable the batching and chunking at the same time. This will throw an exception when initializing the retryLetterProducer:
https://github.com/apache/pulsar/blob/02147454c425b92f0cd1caefa73b9339db6a0269/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java#L96-L97

Reply:
Corrected the test case. Do we need to add validation somewhere else in this changeset, or is having it in the `ProducerBuilderImpl` enough?

Reply:
IMO, we shouldn't expose these configurations to DeadLetterPolicy. This will introduce some producer concepts to the consumer.